src/float16.cr

```diff
@@ -0,0 +1,107 @@
+lib LibIntrinsics
+  fun f16tof32 = "llvm.convert.from.fp16.f32"(Int16) : Float32
```

Comment:
```suggestion
  fun f16tof32 = "llvm.convert.from.fp16.f32"(Float16) : Float32
```

And ditto below?

Reply:
I thought about this, but I wanted to keep the original LLVM API. All of this is then abstracted by `Float16`. Using `Float16` here feels a bit circular dependency to me (which is not, but still...).